docs/virtualserver-and-virtualserverroute.md

```diff
@@ -307,7 +308,39 @@ healthCheck:
 | `read-timeout` | The timeout for reading a response from an upstream server. By default, the `read-timeout` of the upstream is used. | `string` | No |
 | `send-timeout` | The timeout for transmitting a request to an upstream server. By default, the `send-timeout` of the upstream is used. | `string` | No |
 | `headers` | The request headers used for health check requests. NGINX Plus always sets the `Host`, `User-Agent` and `Connection` headers for health check requests. | [`[]header`](#Header) | No |
-| `statusMatch` | The expected response status codes of a health check.  By default, the response should have status code 2xx or 3xx. Examples: `“200”`, `“! 500”`, `"301-303 307"`. See the documentation of the [match](https://nginx.org/en/docs/http/ngx_http_upstream_hc_module.html?#match) directive. | `string` | No |
+| `statusMatch` | The expected response status codes of a health check. By default, the response should have status code 2xx or 3xx. Examples: `“200”`, `“! 500”`, `"301-303 307"`. See the documentation of the [match](https://nginx.org/en/docs/http/ngx_http_upstream_hc_module.html?#match) directive. | `string` | No |
+
+### Upstream.SessionCookie
+
+The SessionCookie field configures session persistence which allows requests from the same client to be passed to the same upstream server. The information about the designated upstream server is passed in a session cookie generated by NGINX Plus.
+
+In the example below, we configure session persistence with a session cookie for an upstream and configure all the available parameters:
+
+```yaml
+name: tea
+service: tea-svc
+port: 80
+sessionCookie:
+  enable: true
+  name: srv_id
+  path: /
+  expires: 1h
+  domain: .example.com
+  httpOnly: false
+  secure: true
+```
+See the [`sticky`](https://nginx.org/en/docs/http/ngx_http_upstream_module.html?#sticky) directive for additional information. The session cookie corresponds to the `sticky cookie` method.
+
+
+| Field | Description | Type | Required |
+| ----- | ----------- | ---- | -------- |
+| `enable` | Enables session persistence with a session cookie for an upstream server. The default is `false`. | `boolean` | No |
+| `name` | The name of the cookie. | `string` | Yes |
+| `path` | The path for which the cookie is set. | `string` | No |
+| `expires` | The time for which a browser should keep the cookie. Can be set to the special value `max`, which will cause the cookie to expire on `31 Dec 2037 23:55:55 GMT`. | `string` | No |
+| `domain` | The domain for which the cookie is set. | `string` | No |
+| `httpOnly` | Adds the `HttpOnly` attribute is added to the cookie. | `boolean` | No |
```

Comment:
This line (description) and the next one seem weird?  `Adds the X attribute is added to the cookie` , shouldn't it be `Adds the X attribute to the cookie` or `X attribute is added to the cookie` ??

Reply:
Yep, doesn't make sense. Changing both to `Adds the X attribute to the cookie`